### DISCUSSION TITLE
Directus EAV

### DISCUSSION THREAD

**Discussion by bridge2web** [directus/directus#2840] [Archive]
How to implement [EAV ](https://en.wikipedia.org/wiki/Entity%E2%80%93attribute%E2%80%93value_model) in Directus?

As a simple example, a _Product_ that has _Attributes_.
These _Attributes_ need the ability to **filter** _Products_.

**Comment by rijkvanzanten** [comment]
Use relationships: https://docs.directus.io/guides/relationships.html

A single product with multiple attributes could be configured as a one-to-many from products->attributes. If an attribute can be used by multiple products, use a many-to-many instead

**Reply by bridge2web** [reply]
Thank you for quick answer! But the question remains how to use the filter.

Products:

![screenshot-delovoibiysk autobo ru-2020 09 11-12_22_07](https://user-images.githubusercontent.com/2793078/92880664-75356180-f438-11ea-812f-b83ae8d70efa.png)

Attributes:

![screenshot-delovoibiysk autobo ru-2020 09 11-12_21_07](https://user-images.githubusercontent.com/2793078/92879014-d78d6280-f436-11ea-9fb6-0e25901c5ef0.png)

How to filter Products with ram = 4 AND color = black?
I try to do this:
`/items/products?filter[attributes.name][eq]=ram&filter[attributes.value][eq]=4&filter[attributes.name][eq]=color&filter[attributes.value][eq]=black`
This is not working properly.

`{"data":[{"id":1,"sku":"notebook-1","price":1000},{"id":2,"sku":"notebook-2","price":1100}],"public":true}`

Maybe I should change the architecture?
